### PR TITLE
Properly handle inlined vertex data

### PR DESCRIPTION
**Pull request by kd-11**
Properly handle inlined vertex data. This should fix missing geometry in alot of games.

### CODE REVIEW

**Comment by kd-11** [comment]
@vlj I noticed when fetching the attributes from the inlined array, dx12 is not taking the initial offset into account.
https://github.com/RPCS3/rpcs3/blob/master/rpcs3/Emu/RSX/D3D12/D3D12Buffer.cpp#L177-Lundefined
i.e if attribs are packed Vertex|Texture|Vertex then Texture stream starts at initial offset element_sizeof(Vertex) and not zero. Taking this into consideration fixes post-processing bugs in outrun online using dx12. I could add it in for dx12, but I'd like your input first.

**Comment by vlj** [comment]
How do you get Outrun to work ? For me it's loading forever atm.

**Comment by vlj** [comment]
I see what you mean, you're indeed correct. If you have a fix then feel free to add it to the PR.

**Comment by kd-11** [comment]
Updated with fix for DX12 as well

File: rpcs3/Emu/RSX/D3D12/D3D12Buffer.cpp [hunk 1]
```diff
@@ -147,8 +147,12 @@ std::tuple<std::vector<D3D12_SHADER_RESOURCE_VIEW_DESC>, size_t> upload_inlined_
 	// We can't rely on vertex_attribute_infos strides here so compute it
 	// assuming all attributes are packed
 	u32 stride = 0;
+	u32 initial_offsets[rsx::limits::vertex_count];
+	u8  index = 0;
```
Comment (danilaml) [inline rpcs3/Emu/RSX/D3D12/D3D12Buffer.cpp]: any reason for u8 over plain int?

Comment (kd-11) [inline rpcs3/Emu/RSX/D3D12/D3D12Buffer.cpp]: range is 0-15

Comment (danilaml) [inline rpcs3/Emu/RSX/D3D12/D3D12Buffer.cpp]: @kd-11 doesn't mean it's better/faster. At leas in any significant way. It'd be probably stored in register anyway.

**Pull request merged**